appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -11,18 +11,23 @@ import { SiteClient } from '../SiteClient';
 import { nonNullProp } from '../utils/nonNull';
 import { formatDeployLog } from './formatDeployLog';
 
-export async function waitForDeploymentToComplete(client: SiteClient, kuduClient: KuduClient, pollingInterval: number = 5000): Promise<void> {
+export async function waitForDeploymentToComplete(client: SiteClient, kuduClient: KuduClient, permanentId?: string, pollingInterval: number = 5000): Promise<void> {
     const alreadyDisplayedLogs: string[] = [];
     let nextTimeToDisplayWaitingLog: number = Date.now();
-    let permanentId: string | undefined;
     let initialStartTime: Date | undefined;
     let deployment: DeployResult | undefined;
+    // a 20 second timeout period to let Kudu initialize the deployment
+    const maxTimeToWaitForPermanentId: number = Date.now() + 20 * 1000;
```

Comment:
Any ideas why I ran into this error before?: https://github.com/Microsoft/vscode-azurefunctions/issues/982

Do we just need to increase the timeout?

Reply:
Hmm, I kind of doubt that it's the timeout.  We're just waiting for the kudu build to initialize so that we can find the deployment id.  I get the feeling that for some reason, the build never initializes.  Not sure if that'd be something on our end or if Kudu never picks up our redeploy request?

Reply:
The only reason I could see it being the timeout is because there is some sort of "temp" phase for deployments that happens after it triggers and before `deployment.id` is the permanent id. I just don't remember what happens in that temp phase or how long it takes

Reply:
Also when I filed that bug - it did successfully redeploy even though I got the timeout error.

Reply:
I'm not 100% sure about this, but I think the "temp" phase is only for new deployments.  However, redeploy already has a `DeployResult` associated with it, so rather than creating a new one, Kudu reuses it to redeploy that deployment. (the game is to see how many times I can say deploy in one sentence)

Reply:
Okay well maybe just try redeploy a few times and time it somehow? (some console.logs maybe?). If it's remotely close to 20 then bump it otherwise I'm fine to leave it as is